### PR TITLE
IAP: Disable all links & buttons that add upgrade to the cart 

### LINKED ISSUE #291: Plans/Domains: Add a feature flag that disables payments in Calypso
We should have one flag that disables all paths to /checkout in Calypso. We could replace these links with information about where to get upgrades.

This will make it possible to list the desktop app in the app store, and still show some information about upgrades.

### PR DESCRIPTION
**Pull request by gziolo**
It partially fixes #291.

Features covered in this PR:
* [x] Domains search
* [x] Renew button
* [x] Credit card list and edit

### Testing
1. Disable the following flags in the config file:
 * `upgrades/checkout`
 * `upgrades/credit-cards`
 * `upgrades/domain-search`
2. Execute `make run` and go to http://calypso.localhost:3000/purchases.
3. Select a Purchase that expires soon.
4. On Manage Purchase page there should be no Renew button.
5. You shouldn't be able to edit Credit Card details.
6. Go to Billing History (http://calypso.localhost:3000/me/billing) and scroll to the bottom of the page.
7. You shouldn't see list of credit cards.
8. Go to Domain (http://calypso.localhost:3000/domains/manage).
9. Select a Site.
10. You shouldn't see any button that allows to add domain.
11. Go to Site Settings (http://calypso.localhost:3000/settings/general).
12. Select a Site.
13. In Site Address section you shouldn't see any button or link which allows to add domain, domain mapping or site redirect.
14. There should be no information about domain registration in Change Site Address.

Now enable flags listed in (1) and repeat all other steps, you should see all action buttons and links :)

### CODE REVIEW

**Comment by gziolo** [comment]
Pinging @scruffian, @janm6k and @peterbutler for review.

**Comment by janm6k** [comment]
I am trying to test the changes in `client/my-sites/site-settings/delete-site-options/index.jsx` - what do I need to have and do to see that component?

**Comment by gziolo** [comment]
> I am trying to test the changes in client/my-sites/site-settings/delete-site-options/index.jsx - what do I need to have and do to see that component?

Point (14) from the list. Yes, I know that the name of the component is a bit tricky :)

**Comment by janm6k** [comment]
Aha :)

In that case, :shipit:

File: client/me/purchases/manage-purchase/index.jsx [hunk 1]
```diff
@@ -155,7 +166,9 @@ const ManagePurchase = React.createClass( {
 									cardExpiry: creditCard.expiryMoment.format( 'MMMM YYYY' )
 								},
 								components: {
-									a: <a href={ paths.editCardDetails( this.props.selectedSite.slug, id, creditCard.id ) } />
+									a: canEditPaymentDetails() ?
```
Comment (drewblaisdell) [inline client/me/purchases/manage-purchase/index.jsx]: This function throws an error if it isn't passed a `purchase` object.

**Pull request merged**